trie/trieStorageManagerInEpoch.go

```diff
@@ -0,0 +1,68 @@
+package trie
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go/common"
+	"github.com/ElrondNetwork/elrond-go/errors"
+)
+
+type trieStorageManagerInEpoch struct {
+	*trieStorageManager
+	mainStorer dbWithGetFromEpoch
+	epoch      uint32
+}
+
+func newTrieStorageManagerInEpoch(storageManager common.StorageManager, epoch uint32) (*trieStorageManagerInEpoch, error) {
+	if check.IfNil(storageManager) {
+		return nil, ErrNilTrieStorage
+	}
+
+	tsm, ok := storageManager.GetBaseTrieStorageManager().(*trieStorageManager)
+	if !ok {
+		storerType := fmt.Sprintf("%T", storageManager.GetBaseTrieStorageManager())
```

Comment:
also here and on L31, could have been a one-liner

Reply:
Done.